components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -457,6 +547,95 @@ - (CGSize)expectedSizeForView:(UIView *)view {
   return expectedItemSize;
 }
 
+/// Sets _selectionIndicator's bounds and center to display under the item at the given index with
+/// no animation. May be called from an animation block to animate the transition.
+- (void)updateSelectionIndicatorToIndex:(NSUInteger)index {
+  if (index == NSNotFound || index >= self.items.count) {
+    // Hide selection indicator.
+    self.selectionIndicatorView.bounds = CGRectZero;
+    return;
+  }
+
+  MDCTabBarViewPositioningTuple *selectedItemViewPositionInContainerView =
+      [self positionInContainerViewForItemAtIndex:index];
+
+  // Place selection indicator under the item's cell.
+  CGRect selectionIndicatorBounds = selectedItemViewPositionInContainerView.bounds;
+  CGPoint selectionIndicatorCenter = selectedItemViewPositionInContainerView.center;
+  self.selectionIndicatorView.bounds = selectionIndicatorBounds;
+  self.selectionIndicatorView.center = [self convertPoint:selectionIndicatorCenter
+                                                 fromView:self.containerView];
+
+  // Extract content frame from item view.
+  CGRect contentFrame = selectionIndicatorBounds;
+  UIView *itemView = self.containerView.arrangedSubviews[index];
+  if ([itemView conformsToProtocol:@protocol(MDCTabBarViewIndicatorSupporting)]) {
+    UIView<MDCTabBarViewIndicatorSupporting> *supportingView =
+        (UIView<MDCTabBarViewIndicatorSupporting> *)itemView;
+    contentFrame = supportingView.contentFrame;
+  }
+
+  // Construct a context object describing the selected tab.
+  UITabBarItem *item = self.items[index];
+  MDCTabBarViewPrivateIndicatorContext *context =
+      [[MDCTabBarViewPrivateIndicatorContext alloc] initWithItem:item
+                                                          bounds:selectionIndicatorBounds
+                                                    contentFrame:contentFrame];
+
+  // Ask the template for attributes.
+  id<MDCTabBarViewIndicatorTemplate> template = self.selectionIndicatorTemplate;
+  MDCTabBarViewIndicatorAttributes *indicatorAttributes =
```

Comment:
Why do we need an attributes when we already have the context? It seems increasing the complication.

Reply:
I agree that in the future it may be an opportunity to reduce the number of classes used to handle the indicator.  However, it hasn't caused problems so far. I'd rather keep what works (in this case) rather than try to refactor/rewrite it. Given our time limitations, it's probably not the best use of our effort.